dev/service-registry/registration-javadsl/src/main/scala/com/lightbend/lagom/internal/server/ServiceRegistrationModule.scala

```diff
@@ -38,12 +38,18 @@ object ServiceRegistrationModule {
       // In dev mode, `play.server.http.address` is used for both HTTP and HTTPS.
       // Reading one value or the other gets the same result.
       val httpAddress = config.getString("play.server.http.address")
-      val uris = List("http", "https").map { scheme =>
-        val port = config.getString(s"play.server.$scheme.port")
-        new URI(s"$scheme://$httpAddress:$port")
+
+      val uris = immutable.Seq.newBuilder[URI]
+
+      val httpPort = config.getString("play.server.http.port")
+      uris += new URI(s"http://$httpAddress:$httpPort")
+
+      if (config.hasPath("play.server.https.port")) {
+        val httpsPort = config.getString("play.server.https.port")
+        uris += new URI(s"https://$httpAddress:$httpsPort")
       }
 
-      ServiceConfig(uris)
+      ServiceConfig(uris.result())
```

Comment:
This change is currently triplicate:

```
$ rg scheme.port | grep -v target | grep -v idea
dev/service-registry/play-integration-javadsl/src/main/scala/com/lightbend/lagom/play/LagomPlayModule.scala:    val port = config.getString(s"play.server.$scheme.port")
dev/service-registry/registration-javadsl/src/main/scala/com/lightbend/lagom/internal/server/ServiceRegistrationModule.scala:        val port = config.getString(s"play.server.$scheme.port")
dev/service-registry/devmode-scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/registry/ServiceRegistration.scala:    val port = config.getString(s"play.server.$scheme.port")
```

😒

Reply:
I meant to say that your change is great but you'll have to copy/paste in 2 other places.

Reply:
Deja-vu: https://github.com/lagom/lagom/pull/1568#discussion_r214759401

Thanks, I'd already forgotten.

Reply:
While we don't refactor and remove duplication we need comments on those three places with this reminder. 😒